kotlin/generate-geodatabase/src/main/java/com/esri/arcgisruntime/sample/generategeodatabase/MainActivity.kt

```diff
@@ -0,0 +1,200 @@
+      graphicsOverlays.add(GraphicsOverlay())
+    }
+  }
+
+  /**
+   * Creates a GenerateGeodatabaseJob and runs it.
+   *
+   * @param view the button which calls this function
+   */
+  fun generateGeodatabase(view: View) {
+    // create a geodatabase sync task and load it
+    val geodatabaseSyncTask = GeodatabaseSyncTask(getString(R.string.wildfire_sync))
+    geodatabaseSyncTask.loadAsync()
+
+    geodatabaseSyncTask.addDoneLoadingListener {
+      // draw a box around the extent
+      mapView.apply {
+        // clear any previous operational layers and graphics
+        map.operationalLayers.clear()
+        graphicsOverlays[0].graphics.clear()
+        // show the extent used as a graphic
+        graphicsOverlays[0].graphics.add(
+          Graphic(
+            visibleArea.extent,
+            SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 5f)
+          )
+        )
+      }
+
+      // create parameters for the job with the return attachments option set to false
+      val parameters = geodatabaseSyncTask
+        .createDefaultGenerateGeodatabaseParametersAsync(mapView.visibleArea.extent).get()
+        .apply { isReturnAttachments = false }
+
+      // create the generate geodatabase job
+      val generateGeodatabaseJob =
+        geodatabaseSyncTask.generateGeodatabase(parameters, localGeodatabasePath)
+
+      // show the job's progress in a dialog
+      val dialog = createProgressDialog(generateGeodatabaseJob)
+      dialog.show()
+      // start the job
+      generateGeodatabaseJob.start()
+      // update progress
+      generateGeodatabaseJob.addProgressChangedListener {
+        dialog.progressBar.progress = generateGeodatabaseJob.progress
+        dialog.progressTextView.text = "$generateGeodatabaseJob.progress%"
+      }
+      // get geodatabase when done
+      generateGeodatabaseJob.addJobDoneListener {
+        // close the progress dialog
+        dialog.dismiss()
+        // load the geodatabase and display its feature tables on the map
+        loadGeodatabase(generateGeodatabaseJob, geodatabaseSyncTask)
+      }
+    }
+  }
+
+  /**
+   * Loads the geodatabase from a GenerateGeodatabaseJob and displays its feature tables on the map.
+   *
+   * @param generateGeodatabaseJob the job which generated this geodatabase
+   * @param geodatabaseSyncTask the GeodatabaseSyncTask which created the job
+   */
+  private fun loadGeodatabase(
+    generateGeodatabaseJob: GenerateGeodatabaseJob,
+    geodatabaseSyncTask: GeodatabaseSyncTask
+  ) {
+    // return if the job failed
+    if (generateGeodatabaseJob.status != Job.Status.SUCCEEDED) {
+      val error = generateGeodatabaseJob.error?.message ?: "Unknown error generating geodatabase"
+      Log.e(TAG, error)
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+      return
+    }
+    // if the job succeeded, load the resulting geodatabase
+    val geodatabase = generateGeodatabaseJob.result.apply { loadAsync() }
```

Comment:
as above, probably best to do starts() and loads() outside of apply blocks since they're not defining behaviour